src/toast-container.tsx

```diff
@@ -1,217 +1,310 @@
-    );
-  }
-
-  renderCenterToasts() {
-    const { toasts } = this.state;
-    let { offset, offsetTop } = this.props;
-    let style: ViewStyle = {
-      top: offsetTop || offset,
-      height: height,
-      width: width,
-      justifyContent: "center",
-      flexDirection: "column-reverse",
+
+    /**
+     * Removes all toasts in stack
+     * @returns void
+     */
+    hideAll = () => {
+        this.setState({
+            foldedToast: {...this.state.foldedToast, open: false},
+            toastsHistory: this.state.toastsHistory.map((t) => ({ ...t, open: false })),
+            unfolded: false
+        });
     };
 
-    const data = toasts.filter((t) => t.placement === "center");
-    const foundToast = data.length > 0;
-
-    if (!foundToast) return null;
-
-    return (
-      <KeyboardAvoidingView
-        behavior={Platform.OS === "ios" ? "position" : undefined}
-        style={[styles.container, style]}
-        pointerEvents="box-none"
-      >
-        {toasts
-          .filter((t) => t.placement === "center")
-          .map((toast) => (
-            <Toast key={toast.id} {...toast} />
-          ))}
-      </KeyboardAvoidingView>
-    );
-  }
-
-  render() {
-    return (
-      <>
-        {this.renderTopToasts()}
-        {this.renderBottomToasts()}
-        {this.renderCenterToasts()}
-      </>
-    );
-  }
+    /**
+     * Check if a toast is currently open
+     * @params id: string
+     * @returns boolean
+     */
+    isOpen = (id: string) => {
+        if(this.state.foldedToast.id === id && this.state.foldedToast.open) return true;
+        else return this.state.toastsHistory.some((t) => t.id === id && t.open);
+    }
+
+    renderBottomToasts() {
+        const { foldedToast, toastsHistory } = this.state;
+        let { offset, offsetBottom } = this.props;
+        let style: ViewStyle = {
+            bottom: offsetBottom || offset,
+            justifyContent: "flex-end",
+            flexDirection: "column",
+        };
+
+        // Single toast with props
+        if (foldedToast !== this.dummyToast || toastsHistory.length > 0) {
+            const toast = foldedToast !== this.dummyToast ? foldedToast : toastsHistory.length > 0 ? toastsHistory[0] : this.dummyToast;
+            // @ts-expect-error
+            const onPress = toastsHistory.length > 0 && foldedToast !== this.dummyToast ? () => this.renderUnfolded() : toast.onPress ? toast.onPress : undefined;
+            const type = toastsHistory.length > 0 && foldedToast !== this.dummyToast ? 'multiple' : undefined;
+
+            return (
+                <KeyboardAvoidingView
+                behavior={Platform.OS === "ios" ? "position" : undefined}
+                pointerEvents="box-none"
+                style={[styles.container, style]}
+                >
+                    <Toast key={toast.id} {...toast} onPress={onPress} type={type} />
+                </KeyboardAvoidingView>
+                );
+        } else return null;
```

Comment:
Don't understand this method, why is it called renderBottomToasts if only one toast is rendered